catchpoint-mixin/panels.libsonnet

```diff
@@ -0,0 +1,418 @@
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      hostsContacted:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Hosts contacted',
+          targets=[t.hostsContacted],
+          description='Number of hosts contacted.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('hosts')
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      cacheAccess:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Cache access',
+          targets=[t.cacheAccess],
+          description='Number of cached elements accessed.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      requestSucessRatio:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Requests success ratio',
+          targets=[t.requestSuccessRatio],
+          description='Success ratio of requests made.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('percentunit')
+        + g.panel.timeSeries.standardOptions.withMax(1)
+        + g.panel.timeSeries.standardOptions.withMin(0)
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      redirections:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Redirects',
+          targets=[t.redirections],
+          description='Number of HTTP redirections encountered.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      contentTypesLoadedBySize:
+        pieChart.new(title='Content types loaded by size')
+        + pieChart.queryOptions.withTargets([t.imageLoadedBySize, t.htmlLoadedBySize, t.cssLoadedBySize, t.scriptLoadedBySize, t.fontLoadedBySize, t.xmlLoadedBySize, t.mediaLoadedBySize])
+        + pieChart.options.legend.withPlacement('right')
+        + pieChart.options.withTooltipMixin({
+          mode: 'multi',
+          sort: 'desc',
+        })
+        + pieChart.panelOptions.withDescription('Size of content loaded.')
+        + pieChart.standardOptions.withUnit('decbytes'),
+
+      contentLoadedByType:
+        barGauge.new(title='Content loaded by type')
+        + barGauge.queryOptions.withTargets([t.imageLoadedByType, t.htmlLoadedByType, t.cssLoadedByType, t.scriptLoadedByType, t.fontLoadedByType, t.xmlLoadedByType, t.mediaLoadedByType])
+        + barGauge.panelOptions.withDescription('Number of elements loaded.')
+        + barGauge.options.withOrientation('horizontal')
+        + barGauge.standardOptions.thresholds.withSteps([
+          barGauge.thresholdStep.withColor('super-light-green'),
+        ]),
+
+      errors:
+        barGauge.new(title='Errors')
+        + barGauge.queryOptions.withTargets([t.objectLoadedError, t.DNSError, t.loadError, t.timeoutError, t.connectionError, t.transactionError])
+        + barGauge.panelOptions.withDescription('Indicates various potential errors that are occuring.')
```

Comment:
Instead of "potential errors" it should be just "errors" since this panel will only show errors that occur and are not bound to occur.